### ISSUE TITLE
[OS X 10.8 Mountain Lion] All tabs crash since WF 56.0.3

### ISSUE THREAD

**Issue by jhzzzzz** [BrowserWorks/waterfox#419]
Hi,
I upgraded Waterfox from 56.0.2 to 56.0.4.1 and the tabs crash.
After experimenting a bit (Safe mode, then new profile), it still happens.
So according to help provided on the reddit forum, I raise this issue.
What kind of information would be useful? I pasted my about:support in the reddit forum, do you want me to post it here too?
Thanks.

**Comment by grahamperrin** [comment]
Thanks, cross reference: 

https://www.reddit.com/r/waterfox/comments/7uvihq/waterfox_5604_is_now_available_for_update_and/dto6ybw/

> Hint: if you disable multiprocess before reproducing a crash, then maybe the browser (not just a tab) will crash. The resulting crash report will be useful. 

## With multi-process disabled in 56.0.4.1

about:preferences#general to disable the feature. 

Can you make Waterfox crash?

If not a crash, then does Waterfox stop responding? https://en.wikipedia.org/wiki/Spinning_pinwheel

**Comment by jhzzzzz** [comment]
I think we may have something.
In the preferences, Multiprocessing is always unchecked. BUT about:support doesn't tell the same story between 56.0.2 (working) and 56.0.4.1 (crashing). 
Compare
Nom: Waterfox Version: 56.0.4.1 Identifiant de compilation: 20180202191140 Canal de mise à jour: release Agent utilisateur: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:56.0) Gecko/20100101 Firefox/56.0.4 Waterfox/56.0.4 Système d’exploitation: Darwin 12.6.0 Fenêtres multi-processus: 1/1 (Activé par l’utilisateur)
.
.
.
with
Nom: Waterfox Version: 56.0.2 Identifiant de compilation: 20180106012217 Canal de mise à jour: release Agent utilisateur: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:56.0) Gecko/20100101 Firefox/56.0.2 Waterfox/56.0.2 Système d’exploitation: Darwin 12.6.0 Fenêtres multi-processus: 0/1 (Désactivé par des modules complémentaires)

Could it be just a problem of multiprocessing being automatically activated in Waterfox > 56.0.2 without the GUI showing it ?

**Comment by jhzzzzz** [comment]
Let me had that, I never had Waterfox itself crash or stop to responding. Only (if one can say that) every tab crashing.
Every tab supposed to display an external page, that is, as all the about:something I tried worked correctly.

**Comment by grahamperrin** [comment]
>> Fenêtres multi-processus: 1/1 (Activé par l’utilisateur)

Confusingly, 56.0.3 makes that statement when the user has **not** enabled multi-process. Related:  

- [Waterfox: e10s, multi-process, extensions.e10sBlocksEnabling and extensions.e10sMultiBlocksEnabling · Issue #397 · MrAlex94/Waterfox](https://github.com/MrAlex94/Waterfox/issues/397)

----

about:preferences#general 

– please go to that address, then manually disable multi-process Waterfox. 

The intention is to crash Waterfox. From <https://www.reddit.com/r/waterfox/comments/7uvihq/waterfox_5604_is_now_available_for_update_and/dtoioxz/>: 

>>> The resulting crash report will be useful.

**Comment by jhzzzzz** [comment]
Okay. No way to crash Waterfox. Each time I check the multiprocessing option, it is automatically unchecked when I come back to preferences.
And the behaviour do not change: each tab crashes but the whole application doesn't.
Anyway, I decided to observe the console when a tab crashes, and each time three messages appear.
The first (full of unicode chars): 
04/02/18 01:33:36,667 ReportCrash[36400]:
Argument num 9 utf8 null: Öû‚Öü‚àï‚à∂‚éÆ‚ï±‚ß∂‚ß∏‚´ª‚´Ω‚ø∞‚ø±‚ø≤‚ø≥‚ø¥‚øµ‚ø∂‚ø∑‚ø∏‚øπ‚ø∫‚øª„ÄÄ„ÄÇ„Äî„Äï„Ä≥„Ç†„Ö§„àù„àû„éÆ„éØ„èÜ„èüÍûâÔ∏îÔ∏ïÔ∏øÔπùÔπûÔªøÔºéÔºèÔΩ°Ôæ†ÔøπÔø∫ÔøªÔøºÔøΩ|153:8;moderate|188:38;{1883021e-e98d-1641-af44-50f7a7fc80fd}|

The second is an indication that a crash report has been created.
Here it is
[crash report for plugin-container [36404].txt](https://github.com/MrAlex94/Waterfox/files/1692349/crash.report.for.plugin-container.36404.txt)

The third is just a message indicating excessive log has been removed (and the reference of a previous crash report which has just been deleted).

Hope this helps...

**Comment by grahamperrin** [comment]
Thanks. Now please visit:

about:plugins

1. save the page in _Web Page, complete_ format
2. compress the file `About Plugins.html` (no need to include the `About Plugins_files` folder)
3. attach the compressed file to this issue. 

(As the crash occurs in safe mode, so we should not suspect a plugin … still, I'm curious.)

----

Also it might be useful to have a System Report file, 

<kbd>Alt</kbd>-Apple menu ▸ System Information

– but please be aware that the report will include identifying information such as the serial number of your Mac.

**Comment by jhzzzzz** [comment]
You stated that as the crash occurs in safe mode, it shouldn't be plugin related, but I'm under the impression that plugins are loaded in safe mode...
Anyway, with WF 56.0.4.1, about:plugins crashes. Both with my test (empty) new profile and my normal profile.
Strangely, I was sure it worked this morning but I can be mistaken.
So here's the about:plugins result, generated from 56.0.2 (he content is obviously the same) and a system information report (short form).
[Archive.zip](https://github.com/MrAlex94/Waterfox/files/1692668/Archive.zip)

Again, thanks for your help.

**Comment by grahamperrin** [comment]
OK, https://helpx.adobe.com/flash-player.html draws attention to 28.0.0.137. 

27.0.0.187 is outdated, please use the Flash Player pane of Préférences Système to check and install an update.  

Here, my installation of Java on Mac OS X 10.9.5 is outdated. I'll update, then restart 56.0.4.1 and tell whether about:plugins leads to a crash.

**Comment by jhzzzzz** [comment]
It seems Flash Player 28 requires macOS 10.9 min.
As for Java, I updated this morning before collecting the data you requested.
It crashed before (no way to be sure about:plugins crashed then, but tab crashing was already there) and it crashes now. So I highly doubt it is Java related.
A better approach could be to list differences between WF 56.0.2 and 56.0.3 because it's probably there that something nefarious for my system has been introduced.

**Comment by jhzzzzz** [comment]
Perusing the release notes, I have a strong feeling the problem is caused by this :
 "Enable e10s, disable e10s-multi for add-ons marked as incompatible. If you have issues with any add-ons, disable e10s completely."

Is there an option in about:config that I can use to ensure that e10s is really totally disabled (I don't really trust the checkmark in Preferences ▸ General)? Is it user-selectable or a compilation option specific to each platform?

**Comment by grahamperrin** [comment]
> … ensure that e10s is really totally disabled  …

1. about:config
2. add a new boolean
3. `browser.tabs.remote.force-disable`
4. `true`
5. quit Waterfox. 

The preference will be effective at the next start.

----

Points of interest:

- a crash with Waterfox in safe mode, which disables add-ons (including plugins – at https://helpx.adobe.com/flash-player.html the _Check Now_ button will not perform a check, et cetera), so I do not imagine a crash caused by an extension that is incompatible with multi-process
- a crash with a new profile
- most remarkable, **a crash at about:plugins** 

From the crash report at <https://github.com/MrAlex94/Waterfox/issues/419#issuecomment-362870235>: 

```text
Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   XUL                           	0x000000010740dc15 0x10589f000 + 28765205
1   XUL                           	0x0000000105fd4e04 0x10589f000 + 7560708
2   XUL                           	0x0000000105d81e25 0x10589f000 + 5123621
3   XUL                           	0x0000000105d8116c 0x10589f000 + 5120364
4   XUL                           	0x0000000105d81a2b 0x10589f000 + 5122603
5   XUL                           	0x0000000105969b1e 0x10589f000 + 830238
6   XUL                           	0x000000010596b4af 0x10589f000 + 836783
7   XUL                           	0x0000000105d840e0 0x10589f000 + 5132512
8   XUL                           	0x0000000105d537c9 0x10589f000 + 4933577
9   XUL                           	0x000000010874a30e 0x10589f000 + 48935694
10  XUL                           	0x0000000105d537c9 0x10589f000 + 4933577
11  XUL                           	0x000000010874a168 0x10589f000 + 48935272
12  org.mozilla.plugincontainer   	0x0000000105301ee2 0x105301000 + 3810
13  org.mozilla.plugincontainer   	0x0000000105301e84 start + 52
```

I suggest: 

1. quit Waterfox
2. for test purposes, uninstall Adobe Flash Player
3. start Waterfox 56.0.4.1
4. visit about:plugins

**Comment by grahamperrin** [comment]
Also please use Utilitaire de disque to tell whether the (HFS Plus) file system _appears_ to be OK (_semble_ être en bon état).

**Comment by jhzzzzz** [comment]
1. My startup disk was corrrupt. After repairing it, no difference for Waterfox.
2. I'm not very confident in removing Flash as I'm afraid the only version downloadable may be 28, which won't run on my Mac.
3. adding the boolean browser.tabs.remote.force-disable with the value true works like a charm (I'm writing this message with WF 56.0.4.1).
This does not explain what causes the crashes but it avoids them.
For me this can be sufficient, but if you want me to try other solutions in order to really understand the matter, I'm willing to do it.

Anyway, many thanks to you

**Comment by grahamperrin** [comment]
Thanks, that's weird! 

Like, working with extensions without multi-process is _less_ crash-prone than (safe mode) all extensions disabled. I never encountered anything like that before.

**Comment by jhzzzzz** [comment]
My understanding is that multiprocessing doesn't work here (is it my Mac? is it Mountain Lion? I cannot tell). As soon as multiprocess is activated, tabs crash (no matter what your extensions are).
Anyway 56.0.2 didn't use multiprocessing and 56.0.3 and 56.0.4 do («out of the box»).

In about:support
56.0.2                  Fenêtres multi-processus: 0/1 (Désactivé par des modules complémentaires)
56.0.4.1 (before ) Fenêtres multi-processus: 1/1 (Activé par l’utilisateur)  // don't know why it was considered activated by user as I had not changed anything AND it was unchecked in the preferences
56.0.4.1 (now)     Fenêtres multi-processus 	0/1 (Désactivé de force)

Do you want me to close the issue of is it useful to keep it open in order to find a correction (now we have a workaround)?

**Comment by grahamperrin** [comment]
> … Fenêtres multi-processus: 1/1 (Activé par l’utilisateur) … it was unchecked in the preferences

That contradiction is weird. It smells like a bug, but I can not reproduce it on 10.9.5: 

- `[  ] Activer le mode multiprocessus de Waterfox` (not ticked, not checked; disabled)
- Fenêtres multi-processus 0/1 (Désactivé) 

<strike>Tomorrow I might have an opportunity to test on Mountain Lion. </strike>I thought that there was a Mac with Mountain Lion handy at work, but there's not.

**Comment by MrAlex94** [comment]
Sorry for missing this thread - e10s doesn't work on older OS X systems, and I did implement a workaround but it only works on first boot, and after a restart or relaunch, gets enabled. I _am_ aware of the issue and am working on a fix.

**Comment by jhzzzzz** [comment]
Thank you.
The workaround, for now, is to force-disable e10s.
I can now use 56.0.4.1.
Keep up the good work!

**Comment by grahamperrin** [comment]
https://github.com/MrAlex94/Waterfox/commit/151f2ba159ca2c6bc11d45045dcb956a7095a9fa noted with thanks, 

>  Block e10s on Yosemite and earlier.

**Comment by grahamperrin** [comment]
#538

**Comment by Peacock365** [comment]
FIXED BY WATERFOX 56.1 DIABLING ELECTROLYSIS FOR MAC OS X 10.7!

**Comment by grahamperrin** [comment]
@MrAlex94 wrote: 

> … e10s doesn't work on older OS X systems, …

<strike>Is that maybe true for 10.9.5 (Mavericks) as well as Mountain Lion? 

(I _do_ occasionally test on Mavericks, but it's nearly always lightweight testing. Not enough to tell where things might fail.)</strike>

Now in Reddit: 

- [e10s: availability of multi-process with older versions of operating systems : waterfox](https://www.reddit.com/r/waterfox/comments/8rtb3l/e10s_availability_of_multiprocess_with_older/)
